### ISSUE TITLE
Unexpected fail when app use async redirect (without window._top)

### ISSUE THREAD

**Issue by njevdjo** [cypress-io/cypress#5731]
Examples
 - This can be login with ajax and redirect to home page .
 - In SPA app add new item with ajax and redirect to detail page.
 or use any async redirection.

Many SPA routers uses similar approach.

### Current behavior:

When in app change hash location with delay (`setTimeout`) test will leave current page and go to parent page (cypress runner page). In my case `http://localhost:8080/__/#/some/route/1`

 - If set "chromeWebSecurity" to `true`, when run `cypress open` and select test get error 
	`CypressError: Cypress detected a cross origin error happened on page load:...`
 - If set "chromeWebSecurity" to `false`, when run `cypress run` get error 
	`Error: EISDIR: illegal operation on a directory, open 'C:\Users\fakeuser\AppData\Roaming\Cypress\cy\production\projects\cypress-playground-98c610aaf979570d4bd341aac2a02314\bundles'`

### Desired behavior:

When in app change hash location with delay (`setTimeout`) test will stay on current page. In my case http://localhost:8080/index.html#/some/route/1

Do not want to see any error.

### Steps to reproduce: (app code and test code)

Added failing test case in forked cypress-test-tiny https://github.com/njevdjo/cypress-test-tiny

#### App code:
```html
<!DOCTYPE html>
<html>

<head>
    <title>Cy issue</title>
    <meta charset="utf-8" />
</head>

<body>
    <div id="my-link">CLICK ME</div>
    <script>
        document.getElementById("my-link").addEventListener("click", () => {
            setTimeout(() => window.location.href = "#/some/route/1", 500)
        })
    </script>
</body>

</html>
```

#### Test code:
```js
describe('cy issue hash with delay', () => {
	beforeEach(function () {
		cy.visit("index.html")
	})

	it('change hash with delay', () => {
		cy.get('#my-link').contains('CLICK ME').click();
		
		cy.wait(1000);

		cy.get('#my-link').contains('CLICK ME')
	})
})
```

### Versions

Cypress package version: 3.6.1
Cypress binary version: 3.6.1
Operating System: Windows10 64bit
Browser: Chrome 78
Electron: 73 (headless)

Related to:
#1244
#1681
#2367
#3077
#3697
#3994
#5376
#5629
#5717

**Comment by njevdjo** [comment]
Workaround by change app code from
```js
setTimeout(() => window.location.href = "#/some/route/1", 500)
```
to
```js
setTimeout(() => window.location.hash = "#/some/route/1", 500)
```

**Comment by jennifer-shehane** [comment]
This is a duplicate of this issue: https://github.com/cypress-io/cypress/issues/3975 Which we are aware of but have had no success fixing.

It'd be great if you could add these details to that issue.

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
